Corona-Warn-App/src/test/java/de/rki/coronawarnapp/vaccination/core/repository/ValueSetsRepositoryTest.kt

```diff
@@ -154,37 +99,15 @@ class ValueSetsRepositoryTest : BaseTest() {
     }
 
     @Test
-    fun `mapping is to stored version is correct`() {
-        valueSetDE.also { it.toStoredVaccinationValueSet().validateValues(it) }
-        valueSetEN.also { it.toStoredVaccinationValueSet().validateValues(it) }
-        emptyValueSetEN.also { it.toStoredVaccinationValueSet().validateValues(it) }
-    }
-
-    @Test
-    fun `clear data of server and local storage`() {
-        createInstance().run {
+    fun `clear data of server and local storage`() = runBlockingTest2(ignoreActive = true) {
+        createInstance(this).run {
             clear()
 
-            coVerify(exactly = 1) {
+            coVerify() {
                 vaccinationServer.clear()
-                valueSetsStorage.clear()
+                latestValueSet.first() shouldBe emptyValueSetEn
+                valueSetsStorage.vaccinationValueSet = emptyValueSetEn
```

Comment:
Shouldn't `clear()` write `null` to storage to delete the data?

Reply:
HotDataFlow doesn't allow nullable values, so I decided to clear the value set by overriding it with an empty value set. I could change the Storage to return an empty value set as default value